### PR TITLE
Adding new fields for 1.22 default compatibility

### PR DESCRIPTION
**Pull request by giacomoch**
For: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/62661

### CODE REVIEW

File: controllers/constant/odlm.go [hunk 1]
```diff
@@ -845,6 +845,11 @@ spec:
                 memory: 512Mi
             logLevel: info
             primaryUpdateStrategy: unsupervised
+            primaryUpdateMethod: switchover
+            enableSuperuserAccess: true
+            replicationSlots:
+              highAvailability:
+                enabled: false
```
Comment (pgodowski) [inline controllers/constant/odlm.go:852]: `enabled: false` should this be updated depending on the profile (1 replica vs multiple replicas)?

**Review by NiharDudam — APPROVED** [review]

**Review by pgodowski — CHANGES_REQUESTED** [review]

File: controllers/constant/odlm.go (hunk 1 above)
Comment (pgodowski) [reply controllers/constant/odlm.go:852]: resolved via slack - the values simply ensure 1.18 vs 1.22 default values compatibility

**Review by pgodowski — APPROVED** [review]

**Comment by pgodowski** [comment]
It looks good to me, but @bitscuit @Daniel-Fan can you pls take a look and merge?
It is safe to merge with EDB 1.18 version being used.

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**